File: api/src/endpoints/accounts/userdata.php
<?php

namespace src\Endpoints\Accounts;

use src\Endpoints\Endpoint;
use src\Database\Database;

class UserData extends Endpoint{

    private $userID;

    public function __construct(){

        $db = new Database("db/db.db");

        $this->validateRequestMethod("GET");

        $jwtToken = $this->validateJWT();
        $this->userID = $jwtToken->sub;

        $this->initialiseSQL();
        $queryResults = $db->executeSQL($this->getSQL(), $this->getSQLParams());

        $this->setData( array(
            "length" => count($queryResults), 
            "message" => "Success",
            "data" => $queryResults
        ));
    }

    protected function initialiseSQL(){
        $sql = "SELECT email, forename, surname FROM users WHERE user_id = :userID";
        $this->setSQL($sql);
        $this->setSQLParams(['userID'=>$this->userID]);
    }
}

File: booking_system/lib/models/user_data.dart

class UserData {
  final String? email;
  final String? forename;
  final String? surname;

  UserData({this.email, this.forename, this.surname});

  factory UserData.fromJson(Map<String, dynamic> json) {
    return UserData(
      email: json['email'] as String?,
      forename: json['forename'] as String?,
      surname: json['surname'] as String?,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'email': email,
      'forename': forename,
      'surname': surname,
    };
  }

}
